### ISSUE TITLE
Concurrency launch: CONCURRENCY_BROWSER definition slightly misleading

### ISSUE THREAD

**Issue by goelayu** [thomasdondorf/puppeteer-cluster#520]
Not sure if others would agree, but I felt the definition of `CONCURRENCY_BROWSER` that every URL is launched in a new browser using the incognito context, seemed misleading to me. 

Technically, it spawn one browser per _worker_, and creates a new incognito context (and a new page) per URL. 

Spawning an entirely new browser per URL, would significantly inflate the loading cost (in terms of CPU usage) of each URL (as per some of my experiments).

**Comment by tehandyb** [comment]
It says CONCURRENCY_BROWSER launches one browser per url, and CONCURRENCY_CONTEXT launches an incognito page per url. So if what you're saying is true, and it only spawns one browser per worker, I would agree the docs should be updated :)

**Comment by goelayu** [comment]
Yes, CONCURRENCY_BROWSER launches one browser per worker, and not per URL. @thomasdondorf feel free to update the READme if you agree.

**Comment by localhost-MouhannadlrX** [comment]
Agree with you @goelayu
